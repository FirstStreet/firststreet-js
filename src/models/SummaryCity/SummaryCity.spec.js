const City = require('./index');
const summaryCityMock = require('../../__mocks__/summaryCity');

let summary = null;
describe('City', () => {
  beforeEach(() => {
    summary = new City(summaryCityMock);
  ***REMOVED***);

  it('should initialize', () => {
    expect(summary).toMatchSnapshot();
  ***REMOVED***);

  it('should have FSID', () => {
    expect(summary.FSID).toBeDefined();
  ***REMOVED***);

  it('should have name', () => {
    expect(summary.name).toBeDefined();
  ***REMOVED***);

  it('should have state', () => {
    expect(summary.state).toBeDefined();
  ***REMOVED***);

  it('should have polygon data', () => {
    expect(summary.polygon).toBeDefined();
  ***REMOVED***);

  it('should have polygon geoJSON', () => {
    expect(summary.polygonGeoJSON).toBeDefined();
  ***REMOVED***);

  it('should have center data', () => {
    expect(summary.center).toBeDefined();
  ***REMOVED***);

  it('should have center geoJSON', () => {
    expect(summary.centerGeoJSON).toBeDefined();
  ***REMOVED***);

  it('should have bounds data', () => {
    expect(summary.bounds).toBeDefined();
  ***REMOVED***);

  it('should have bounds geoJSON', () => {
    expect(summary.boundsGeoJSON).toBeDefined();
  ***REMOVED***);
***REMOVED***);
