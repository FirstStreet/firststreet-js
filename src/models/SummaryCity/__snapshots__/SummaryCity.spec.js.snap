// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City should initialize 1`] = `
City {
  "data": Object {
    "FSID": 1222175,
    "results": Object {
      "firstFloodRisk": null,
      "floodRisks": null,
      "location": Object {
        "FSID": 1222175,
        "geometry": Object {
          "bbox": Object {
            "coordinates": Array [
              Array [
                Array [
                  -81.483409,
                  30.599751,
                ],
                Array [
                  -81.425846,
                  30.599751,
                ],
                Array [
                  -81.425846,
                  30.705699,
                ],
                Array [
                  -81.483409,
                  30.705699,
                ],
                Array [
                  -81.483409,
                  30.599751,
                ],
              ],
            ],
            "type": "Polygon",
          },
          "center": Object {
            "coordinates": Array [
              -81.4499316280516,
              30.6589162194187,
            ],
            "type": "Point",
          },
          "polygon": Object {
            "coordinates": Array [
              Array [
                Array [
                  Array [
                    -81.483002,
                    30.663172,
                  ],
                  Array [
                    -81.483409,
                    30.661769,
                  ],
                  Array [
                    -81.482518,
                    30.657585,
                  ],
                  Array [
                    -81.468806,
                    30.656755,
                  ],
                  Array [
                    -81.46511,
                    30.656069,
                  ],
                  Array [
                    -81.45848,
                    30.65537,
                  ],
                  Array [
                    -81.45857,
                    30.6541,
                  ],
                  Array [
                    -81.45742,
                    30.65396,
                  ],
                  Array [
                    -81.45716,
                    30.65522,
                  ],
                  Array [
                    -81.4537,
                    30.65481,
                  ],
                  Array [
                    -81.453825,
                    30.654032,
                  ],
                  Array [
                    -81.45209,
                    30.65462,
                  ],
                  Array [
                    -81.44574,
                    30.6539,
                  ],
                  Array [
                    -81.442727,
                    30.653365,
                  ],
                  Array [
                    -81.443947,
                    30.64894,
                  ],
                  Array [
                    -81.44547,
                    30.646003,
                  ],
                  Array [
                    -81.443612,
                    30.644717,
                  ],
                  Array [
                    -81.443792,
                    30.639722,
                  ],
                  Array [
                    -81.442541,
                    30.639469,
                  ],
                  Array [
                    -81.442767,
                    30.638371,
                  ],
                  Array [
                    -81.445029,
                    30.638199,
                  ],
                  Array [
                    -81.44561,
                    30.640023,
                  ],
                  Array [
                    -81.444992,
                    30.643798,
                  ],
                  Array [
                    -81.446565,
                    30.641326,
                  ],
                  Array [
                    -81.449156,
                    30.641245,
                  ],
                  Array [
                    -81.44912,
                    30.64249,
                  ],
                  Array [
                    -81.44776,
                    30.64473,
                  ],
                  Array [
                    -81.446676,
                    30.649374,
                  ],
                  Array [
                    -81.450715,
                    30.649368,
                  ],
                  Array [
                    -81.450211,
                    30.65284,
                  ],
                  Array [
                    -81.453042,
                    30.652769,
                  ],
                  Array [
                    -81.452146,
                    30.650863,
                  ],
                  Array [
                    -81.454503,
                    30.649731,
                  ],
                  Array [
                    -81.454357,
                    30.650726,
                  ],
                  Array [
                    -81.456973,
                    30.649902,
                  ],
                  Array [
                    -81.457137,
                    30.651196,
                  ],
                  Array [
                    -81.460155,
                    30.65102,
                  ],
                  Array [
                    -81.459331,
                    30.645982,
                  ],
                  Array [
                    -81.457899,
                    30.645578,
                  ],
                  Array [
                    -81.4551,
                    30.64578,
                  ],
                  Array [
                    -81.455403,
                    30.643632,
                  ],
                  Array [
                    -81.456203,
                    30.643353,
                  ],
                  Array [
                    -81.455707,
                    30.643921,
                  ],
                  Array [
                    -81.458969,
                    30.643748,
                  ],
                  Array [
                    -81.458588,
                    30.641409,
                  ],
                  Array [
                    -81.456995,
                    30.641503,
                  ],
                  Array [
                    -81.457182,
                    30.641942,
                  ],
                  Array [
                    -81.455633,
                    30.64192,
                  ],
                  Array [
                    -81.45595,
                    30.63946,
                  ],
                  Array [
                    -81.45589,
                    30.6386,
                  ],
                  Array [
                    -81.4533,
                    30.63879,
                  ],
                  Array [
                    -81.452733,
                    30.633326,
                  ],
                  Array [
                    -81.44458,
                    30.633286,
                  ],
                  Array [
                    -81.442653,
                    30.63467,
                  ],
                  Array [
                    -81.441265,
                    30.634554,
                  ],
                  Array [
                    -81.44207,
                    30.633346,
                  ],
                  Array [
                    -81.443583,
                    30.62339,
                  ],
                  Array [
                    -81.443247,
                    30.620574,
                  ],
                  Array [
                    -81.44254,
                    30.619702,
                  ],
                  Array [
                    -81.449319,
                    30.619755,
                  ],
                  Array [
                    -81.449695,
                    30.62335,
                  ],
                  Array [
                    -81.454646,
                    30.623345,
                  ],
                  Array [
                    -81.454667,
                    30.620697,
                  ],
                  Array [
                    -81.4553,
                    30.620674,
                  ],
                  Array [
                    -81.457071,
                    30.620694,
                  ],
                  Array [
                    -81.45722,
                    30.619736,
                  ],
                  Array [
                    -81.459883,
                    30.619624,
                  ],
                  Array [
                    -81.457217,
                    30.62252,
                  ],
                  Array [
                    -81.460655,
                    30.62248,
                  ],
                  Array [
                    -81.460666,
                    30.625385,
                  ],
                  Array [
                    -81.460054,
                    30.627296,
                  ],
                  Array [
                    -81.462631,
                    30.627342,
                  ],
                  Array [
                    -81.463831,
                    30.630038,
                  ],
                  Array [
                    -81.464238,
                    30.62545,
                  ],
                  Array [
                    -81.46606,
                    30.62652,
                  ],
                  Array [
                    -81.46609,
                    30.62885,
                  ],
                  Array [
                    -81.468454,
                    30.628849,
                  ],
                  Array [
                    -81.468454,
                    30.625215,
                  ],
                  Array [
                    -81.466047,
                    30.625771,
                  ],
                  Array [
                    -81.46604,
                    30.62337,
                  ],
                  Array [
                    -81.462733,
                    30.623357,
                  ],
                  Array [
                    -81.462735,
                    30.621222,
                  ],
                  Array [
                    -81.461116,
                    30.620039,
                  ],
                  Array [
                    -81.466007,
                    30.620138,
                  ],
                  Array [
                    -81.466017,
                    30.620994,
                  ],
                  Array [
                    -81.468923,
                    30.62094,
                  ],
                  Array [
                    -81.470466,
                    30.619894,
                  ],
                  Array [
                    -81.470948,
                    30.61787,
                  ],
                  Array [
                    -81.472096,
                    30.616176,
                  ],
                  Array [
                    -81.470412,
                    30.611868,
                  ],
                  Array [
                    -81.469161,
                    30.611093,
                  ],
                  Array [
                    -81.469989,
                    30.609123,
                  ],
                  Array [
                    -81.46917,
                    30.606181,
                  ],
                  Array [
                    -81.467599,
                    30.603838,
                  ],
                  Array [
                    -81.466612,
                    30.603592,
                  ],
                  Array [
                    -81.468162,
                    30.601963,
                  ],
                  Array [
                    -81.467252,
                    30.600505,
                  ],
                  Array [
                    -81.465139,
                    30.599805,
                  ],
                  Array [
                    -81.459443,
                    30.599751,
                  ],
                  Array [
                    -81.448506,
                    30.599892,
                  ],
                  Array [
                    -81.447085,
                    30.607495,
                  ],
                  Array [
                    -81.44714,
                    30.609113,
                  ],
                  Array [
                    -81.445063,
                    30.609328,
                  ],
                  Array [
                    -81.444982,
                    30.610104,
                  ],
                  Array [
                    -81.443091,
                    30.610258,
                  ],
                  Array [
                    -81.443308,
                    30.608785,
                  ],
                  Array [
                    -81.443572,
                    30.605822,
                  ],
                  Array [
                    -81.443172,
                    30.605879,
                  ],
                  Array [
                    -81.4411254142,
                    30.6060875019,
                  ],
                  Array [
                    -81.437023,
                    30.631352,
                  ],
                  Array [
                    -81.4355199998,
                    30.6385659696,
                  ],
                  Array [
                    -81.432145,
                    30.654765,
                  ],
                  Array [
                    -81.4298328133,
                    30.6672551662,
                  ],
                  Array [
                    -81.429174,
                    30.670814,
                  ],
                  Array [
                    -81.427866,
                    30.683693,
                  ],
                  Array [
                    -81.427599,
                    30.692279,
                  ],
                  Array [
                    -81.425846,
                    30.700452,
                  ],
                  Array [
                    -81.4269508304,
                    30.7005411258,
                  ],
                  Array [
                    -81.429193,
                    30.700722,
                  ],
                  Array [
                    -81.432725,
                    30.703017,
                  ],
                  Array [
                    -81.438045,
                    30.705045,
                  ],
                  Array [
                    -81.443342,
                    30.703935,
                  ],
                  Array [
                    -81.447698,
                    30.705007,
                  ],
                  Array [
                    -81.454121,
                    30.705699,
                  ],
                  Array [
                    -81.458614,
                    30.703486,
                  ],
                  Array [
                    -81.460013,
                    30.703308,
                  ],
                  Array [
                    -81.460926,
                    30.701463,
                  ],
                  Array [
                    -81.461001,
                    30.697156,
                  ],
                  Array [
                    -81.457968,
                    30.693996,
                  ],
                  Array [
                    -81.45343,
                    30.69476,
                  ],
                  Array [
                    -81.45182,
                    30.69378,
                  ],
                  Array [
                    -81.452713,
                    30.692065,
                  ],
                  Array [
                    -81.456276,
                    30.691485,
                  ],
                  Array [
                    -81.457848,
                    30.690655,
                  ],
                  Array [
                    -81.457269,
                    30.690495,
                  ],
                  Array [
                    -81.45653,
                    30.686496,
                  ],
                  Array [
                    -81.458064,
                    30.686353,
                  ],
                  Array [
                    -81.458678,
                    30.684395,
                  ],
                  Array [
                    -81.461951,
                    30.690192,
                  ],
                  Array [
                    -81.464273,
                    30.68507,
                  ],
                  Array [
                    -81.466341,
                    30.682806,
                  ],
                  Array [
                    -81.471252,
                    30.6767,
                  ],
                  Array [
                    -81.473291,
                    30.675695,
                  ],
                  Array [
                    -81.476799,
                    30.67479,
                  ],
                  Array [
                    -81.476846,
                    30.67189,
                  ],
                  Array [
                    -81.475008,
                    30.67219,
                  ],
                  Array [
                    -81.47425,
                    30.670832,
                  ],
                  Array [
                    -81.472398,
                    30.669914,
                  ],
                  Array [
                    -81.474802,
                    30.668197,
                  ],
                  Array [
                    -81.47799,
                    30.666857,
                  ],
                  Array [
                    -81.473546,
                    30.665948,
                  ],
                  Array [
                    -81.469539,
                    30.66476,
                  ],
                  Array [
                    -81.472612,
                    30.663571,
                  ],
                  Array [
                    -81.474574,
                    30.661159,
                  ],
                  Array [
                    -81.475474,
                    30.660754,
                  ],
                  Array [
                    -81.478547,
                    30.661276,
                  ],
                  Array [
                    -81.480031,
                    30.662931,
                  ],
                  Array [
                    -81.482112,
                    30.663876,
                  ],
                  Array [
                    -81.483002,
                    30.663172,
                  ],
                ],
              ],
            ],
            "type": "MultiPolygon",
          },
        },
        "name": "Fernandina Beach",
        "state": "Florida",
      },
    },
  },
  "location": Object {
    "FSID": 1222175,
    "geometry": Object {
      "bbox": Object {
        "coordinates": Array [
          Array [
            Array [
              -81.483409,
              30.599751,
            ],
            Array [
              -81.425846,
              30.599751,
            ],
            Array [
              -81.425846,
              30.705699,
            ],
            Array [
              -81.483409,
              30.705699,
            ],
            Array [
              -81.483409,
              30.599751,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "center": Object {
        "coordinates": Array [
          -81.4499316280516,
          30.6589162194187,
        ],
        "type": "Point",
      },
      "polygon": Object {
        "coordinates": Array [
          Array [
            Array [
              Array [
                -81.483002,
                30.663172,
              ],
              Array [
                -81.483409,
                30.661769,
              ],
              Array [
                -81.482518,
                30.657585,
              ],
              Array [
                -81.468806,
                30.656755,
              ],
              Array [
                -81.46511,
                30.656069,
              ],
              Array [
                -81.45848,
                30.65537,
              ],
              Array [
                -81.45857,
                30.6541,
              ],
              Array [
                -81.45742,
                30.65396,
              ],
              Array [
                -81.45716,
                30.65522,
              ],
              Array [
                -81.4537,
                30.65481,
              ],
              Array [
                -81.453825,
                30.654032,
              ],
              Array [
                -81.45209,
                30.65462,
              ],
              Array [
                -81.44574,
                30.6539,
              ],
              Array [
                -81.442727,
                30.653365,
              ],
              Array [
                -81.443947,
                30.64894,
              ],
              Array [
                -81.44547,
                30.646003,
              ],
              Array [
                -81.443612,
                30.644717,
              ],
              Array [
                -81.443792,
                30.639722,
              ],
              Array [
                -81.442541,
                30.639469,
              ],
              Array [
                -81.442767,
                30.638371,
              ],
              Array [
                -81.445029,
                30.638199,
              ],
              Array [
                -81.44561,
                30.640023,
              ],
              Array [
                -81.444992,
                30.643798,
              ],
              Array [
                -81.446565,
                30.641326,
              ],
              Array [
                -81.449156,
                30.641245,
              ],
              Array [
                -81.44912,
                30.64249,
              ],
              Array [
                -81.44776,
                30.64473,
              ],
              Array [
                -81.446676,
                30.649374,
              ],
              Array [
                -81.450715,
                30.649368,
              ],
              Array [
                -81.450211,
                30.65284,
              ],
              Array [
                -81.453042,
                30.652769,
              ],
              Array [
                -81.452146,
                30.650863,
              ],
              Array [
                -81.454503,
                30.649731,
              ],
              Array [
                -81.454357,
                30.650726,
              ],
              Array [
                -81.456973,
                30.649902,
              ],
              Array [
                -81.457137,
                30.651196,
              ],
              Array [
                -81.460155,
                30.65102,
              ],
              Array [
                -81.459331,
                30.645982,
              ],
              Array [
                -81.457899,
                30.645578,
              ],
              Array [
                -81.4551,
                30.64578,
              ],
              Array [
                -81.455403,
                30.643632,
              ],
              Array [
                -81.456203,
                30.643353,
              ],
              Array [
                -81.455707,
                30.643921,
              ],
              Array [
                -81.458969,
                30.643748,
              ],
              Array [
                -81.458588,
                30.641409,
              ],
              Array [
                -81.456995,
                30.641503,
              ],
              Array [
                -81.457182,
                30.641942,
              ],
              Array [
                -81.455633,
                30.64192,
              ],
              Array [
                -81.45595,
                30.63946,
              ],
              Array [
                -81.45589,
                30.6386,
              ],
              Array [
                -81.4533,
                30.63879,
              ],
              Array [
                -81.452733,
                30.633326,
              ],
              Array [
                -81.44458,
                30.633286,
              ],
              Array [
                -81.442653,
                30.63467,
              ],
              Array [
                -81.441265,
                30.634554,
              ],
              Array [
                -81.44207,
                30.633346,
              ],
              Array [
                -81.443583,
                30.62339,
              ],
              Array [
                -81.443247,
                30.620574,
              ],
              Array [
                -81.44254,
                30.619702,
              ],
              Array [
                -81.449319,
                30.619755,
              ],
              Array [
                -81.449695,
                30.62335,
              ],
              Array [
                -81.454646,
                30.623345,
              ],
              Array [
                -81.454667,
                30.620697,
              ],
              Array [
                -81.4553,
                30.620674,
              ],
              Array [
                -81.457071,
                30.620694,
              ],
              Array [
                -81.45722,
                30.619736,
              ],
              Array [
                -81.459883,
                30.619624,
              ],
              Array [
                -81.457217,
                30.62252,
              ],
              Array [
                -81.460655,
                30.62248,
              ],
              Array [
                -81.460666,
                30.625385,
              ],
              Array [
                -81.460054,
                30.627296,
              ],
              Array [
                -81.462631,
                30.627342,
              ],
              Array [
                -81.463831,
                30.630038,
              ],
              Array [
                -81.464238,
                30.62545,
              ],
              Array [
                -81.46606,
                30.62652,
              ],
              Array [
                -81.46609,
                30.62885,
              ],
              Array [
                -81.468454,
                30.628849,
              ],
              Array [
                -81.468454,
                30.625215,
              ],
              Array [
                -81.466047,
                30.625771,
              ],
              Array [
                -81.46604,
                30.62337,
              ],
              Array [
                -81.462733,
                30.623357,
              ],
              Array [
                -81.462735,
                30.621222,
              ],
              Array [
                -81.461116,
                30.620039,
              ],
              Array [
                -81.466007,
                30.620138,
              ],
              Array [
                -81.466017,
                30.620994,
              ],
              Array [
                -81.468923,
                30.62094,
              ],
              Array [
                -81.470466,
                30.619894,
              ],
              Array [
                -81.470948,
                30.61787,
              ],
              Array [
                -81.472096,
                30.616176,
              ],
              Array [
                -81.470412,
                30.611868,
              ],
              Array [
                -81.469161,
                30.611093,
              ],
              Array [
                -81.469989,
                30.609123,
              ],
              Array [
                -81.46917,
                30.606181,
              ],
              Array [
                -81.467599,
                30.603838,
              ],
              Array [
                -81.466612,
                30.603592,
              ],
              Array [
                -81.468162,
                30.601963,
              ],
              Array [
                -81.467252,
                30.600505,
              ],
              Array [
                -81.465139,
                30.599805,
              ],
              Array [
                -81.459443,
                30.599751,
              ],
              Array [
                -81.448506,
                30.599892,
              ],
              Array [
                -81.447085,
                30.607495,
              ],
              Array [
                -81.44714,
                30.609113,
              ],
              Array [
                -81.445063,
                30.609328,
              ],
              Array [
                -81.444982,
                30.610104,
              ],
              Array [
                -81.443091,
                30.610258,
              ],
              Array [
                -81.443308,
                30.608785,
              ],
              Array [
                -81.443572,
                30.605822,
              ],
              Array [
                -81.443172,
                30.605879,
              ],
              Array [
                -81.4411254142,
                30.6060875019,
              ],
              Array [
                -81.437023,
                30.631352,
              ],
              Array [
                -81.4355199998,
                30.6385659696,
              ],
              Array [
                -81.432145,
                30.654765,
              ],
              Array [
                -81.4298328133,
                30.6672551662,
              ],
              Array [
                -81.429174,
                30.670814,
              ],
              Array [
                -81.427866,
                30.683693,
              ],
              Array [
                -81.427599,
                30.692279,
              ],
              Array [
                -81.425846,
                30.700452,
              ],
              Array [
                -81.4269508304,
                30.7005411258,
              ],
              Array [
                -81.429193,
                30.700722,
              ],
              Array [
                -81.432725,
                30.703017,
              ],
              Array [
                -81.438045,
                30.705045,
              ],
              Array [
                -81.443342,
                30.703935,
              ],
              Array [
                -81.447698,
                30.705007,
              ],
              Array [
                -81.454121,
                30.705699,
              ],
              Array [
                -81.458614,
                30.703486,
              ],
              Array [
                -81.460013,
                30.703308,
              ],
              Array [
                -81.460926,
                30.701463,
              ],
              Array [
                -81.461001,
                30.697156,
              ],
              Array [
                -81.457968,
                30.693996,
              ],
              Array [
                -81.45343,
                30.69476,
              ],
              Array [
                -81.45182,
                30.69378,
              ],
              Array [
                -81.452713,
                30.692065,
              ],
              Array [
                -81.456276,
                30.691485,
              ],
              Array [
                -81.457848,
                30.690655,
              ],
              Array [
                -81.457269,
                30.690495,
              ],
              Array [
                -81.45653,
                30.686496,
              ],
              Array [
                -81.458064,
                30.686353,
              ],
              Array [
                -81.458678,
                30.684395,
              ],
              Array [
                -81.461951,
                30.690192,
              ],
              Array [
                -81.464273,
                30.68507,
              ],
              Array [
                -81.466341,
                30.682806,
              ],
              Array [
                -81.471252,
                30.6767,
              ],
              Array [
                -81.473291,
                30.675695,
              ],
              Array [
                -81.476799,
                30.67479,
              ],
              Array [
                -81.476846,
                30.67189,
              ],
              Array [
                -81.475008,
                30.67219,
              ],
              Array [
                -81.47425,
                30.670832,
              ],
              Array [
                -81.472398,
                30.669914,
              ],
              Array [
                -81.474802,
                30.668197,
              ],
              Array [
                -81.47799,
                30.666857,
              ],
              Array [
                -81.473546,
                30.665948,
              ],
              Array [
                -81.469539,
                30.66476,
              ],
              Array [
                -81.472612,
                30.663571,
              ],
              Array [
                -81.474574,
                30.661159,
              ],
              Array [
                -81.475474,
                30.660754,
              ],
              Array [
                -81.478547,
                30.661276,
              ],
              Array [
                -81.480031,
                30.662931,
              ],
              Array [
                -81.482112,
                30.663876,
              ],
              Array [
                -81.483002,
                30.663172,
              ],
            ],
          ],
        ],
        "type": "MultiPolygon",
      },
    },
    "name": "Fernandina Beach",
    "state": "Florida",
  },
}
`;
